packages/react-aria-components/src/Table.tsx

```diff
@@ -368,7 +368,7 @@ export function useTableOptions(): TableOptionsContextValue {
   return useContext(TableOptionsContext)!;
 }
 
-export interface TableHeaderProps<T> {
+export interface TableHeaderProps<T> extends StyleRenderProps<T> {
```

Comment:
I think this should just extend StyleProps. Otherwise this would have to support a function with T as the argument which doesn't really seem that useful.